src/Controller/IslandoraWorkbenchIntegrationFileRenameController.php

```diff
@@ -0,0 +1,181 @@
+use Symfony\Component\HttpFoundation\Request;
+use Drupal\file\Entity\File;
+use Drupal\Core\File\FileSystemInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Drupal\Core\File\Event\FileUploadSanitizeNameEvent;
+use Symfony\Component\EventDispatcher\EventDispatcherInterface;
+
+class IslandoraWorkbenchIntegrationFileRenameController extends ControllerBase
+{
+
+
+    /**
+     * FileSystem service.
+     *
+     * @var \Drupal\Core\File\FileSystemInterface
+     */
+    protected $fileSystem;
+
+    /**
+     * EventDispatcher service.
+     *
+     * @var \Symfony\Component\EventDispatcher\EventDispatcher
+     */
+    protected $eventDispatcher;
+
+
+    /**
+     * Constructor to inject the Symfony Filesystem service.
+     */
+    public function __construct(FileSystemInterface $fileSystem, EventDispatcherInterface $event_dispatcher)
+    {
+        $this->fileSystem = $fileSystem;
+        $this->eventDispatcher = $event_dispatcher;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public static function create(ContainerInterface $container)
+    {
+        return new static(
+            $container->get('file_system'),
+            $container->get('event_dispatcher')
+        );
+    }
+
+
+
+    /**
+     * Renames a file while keeping the extension unchanged.
+     *
+     * Expects a URL parameter for the file ID (fid) and a JSON payload with:
+     * - new_filename: The new base file name (without extension).
+     *
+     * @param \Symfony\Component\HttpFoundation\Request $request
+     *   The incoming request.
+     * @param Drupal\file\Entity\File                   $file
+     *   The file ID from the URL.
+     *
+     * @return \Symfony\Component\HttpFoundation\JsonResponse
+     *   The JSON response.
+     */
+    public function main(Request $request, $file)
+    {
+        // Decode JSON payload.
+
+        $data = json_decode($request->getContent(), true);
+
+        // Validate that new_filename is provided.
+        if (empty($data['new_filename'])) {
+            return new JsonResponse(['error' => 'Missing new_filename parameter']);
+        }
+
+
+        if (!$file) {
+            return new JsonResponse(["Error 404" => 'No file found']);
+        }
+
+
+        $pathinfo = pathinfo($file->getFileUri());
+        $filename_new = $data['new_filename'];
+        $msg = $this->validate($file, $filename_new);
+        if ($msg != 'OK') {
+            return new JsonResponse(['Error' => $msg]);
+        }
+
+
+
+        if ($filename_new != $file->getFilename()) {
+            $filepath_new = $this->getRenamedFilePath($file, $filename_new);
+            rename($file->getFileUri(), $filepath_new);
+            // Update file entity.
+            $file->setFilename($filename_new);
+            $file->setFileUri($filepath_new);
+            $status = $file->save();
+
+            return new JsonResponse(['message' => 'File renamed successfully', 'new_filename' => $filename_new]);
```

Comment:
This message should contain both the original and new filename. Also, can we add this entry to the Drupal logger?